### PR TITLE
Fix simulation.clone and entity.clone

### PR DESCRIPTION
**Pull request by fpagnoux**
#### Bug fix

- Fix `simulation.clone` and `entity.clone` methods

### CODE REVIEW

**Comment by fpagnoux** [comment]
Fixes france's build issue : https://github.com/openfisca/openfisca-france/pull/786

File: openfisca_core/holders.py [hunk 1]
```diff
@@ -100,7 +100,7 @@ def calculate(self, period, **parameters):
     def calculate_output(self, period):
         return self.formula.calculate_output(period)
 
-    def clone(self):
+    def clone(self, new_entity):
```
Comment (sandcha) [inline openfisca_core/holders.py]: Rename `new_entity` to `entity` ? `new_entity` name assumes something out of the bounds of this method.

File: openfisca_core/simulations.py [hunk 2]
```diff
@@ -132,12 +132,13 @@ def clone(self, debug = False, debug_all = False, trace = False):
...
                 new_dict[key] = value
 
-        new.persons = self.persons.clone()
+        new.persons = self.persons.clone(new)
         setattr(new, new.persons.key, new.persons)
         new.entities = {new.persons.key: new.persons}
 
         for entity_class in self.tax_benefit_system.group_entities:
-            entity = self.entities[entity_class.key].clone()
+            entity = self.entities[entity_class.key].clone(new)
```
Comment (fpagnoux) [inline openfisca_core/simulations.py]: No, it's not a class, but an instance ;)

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## 14.1.1 - [#533](https://github.com/openfisca/openfisca-core/pull/533)
+
+#### Bug fix
+
+- Fix `simulation.clone` and `entity.clone` methods
```
Comment (fpagnoux) [inline CHANGELOG.md]: 👍

File: tests/core/test_simulations.py [hunk 4]
```diff
@@ -25,3 +26,26 @@ def test_calculate__holder_attribute_content():
...
+                "bill": {"salary": {"2017-01": 3000}},
+                },
+            })
+
+    simulation_clone = simulation.clone()
+    assert simulation != simulation_clone
+
+    for entity_id, entity in simulation.entities.iteritems():
+        assert entity != simulation_clone.entities[entity_id]
+
```
Comment (fpagnoux) [inline tests/core/test_simulations.py]: 👌

File: openfisca_core/holders.py (hunk 1 above)
Comment (fpagnoux) [inline openfisca_core/holders.py]: 👌

File: openfisca_core/simulations.py [hunk 5]
```diff
@@ -132,12 +132,13 @@ def clone(self, debug = False, debug_all = False, trace = False):
             if key not in ('debug', 'debug_all', 'trace', 'entities'):
```
Comment (fpagnoux) [inline openfisca_core/simulations.py]: Not sure, but I think we don't want the `debug`, `debug_all` and `trace` attributes of the original simulation to be copied to the new one. Instead, we only take their value from the parameters of the `clone` method.

Comment (fpagnoux) [inline openfisca_core/simulations.py]: >What about the traceback ? Add it to the keys to ignore ?

It was not ignored before this PR, nor before `14.1.0`. As we don't fully understand the subtlety of this code, I'd prefer to try adding things without exactly knowing why.

Comment (fpagnoux) [inline openfisca_core/simulations.py]: > Add `persons` ?

We redefine `new.person` just further, so it's useless. I remove `entities` as well, for the same reason.

**Pull request merged**